### DISCUSSION TITLE
Side bar does not automatically generate.

### DISCUSSION THREAD

**Discussion by thijsmakes** [vuepress/core#334] [Q&A] [answered]
Hi all, 

I can't seem to figure out how I get the sidebar to be automatically generated. The Docs say this is the default behavior. I also tried to force it with `---
sidebar: auto
---` 
But the sidebar will only render 1 README.md file at the root of the /Docs folder. I want to render all the .md files automatically. 

Rendered sidebar:
![image](https://user-images.githubusercontent.com/10755942/127850727-830c7898-140e-4316-8621-69bce20ec822.png) 

Folder structure:
![image](https://user-images.githubusercontent.com/10755942/127850426-a1a603c0-6536-49d8-8f25-dedbdb82afeb.png)

**Comment by meteorlxy** [comment, marked as answer]
`sidebar: auto` is for generating sidebar from all your sub headers in current page, which won't generate links to other pages.

If you do not want to configure sidebar manually, you can write a function to read directory and generate the sidebar config.

**Reply by seangwright** [reply]
Does this function need to generate an object that is the value of `sidebar: ...` in `config.js` or would this customization be done as a hook into the app via a plugin?

**Comment by Mister-Hope** [comment]
If you do want your sidebar be auto generated from file structure, you can search through 3rd party themes and plugins.

BTW I have plans to achieve it in my own theme vuepress-theme-hope.
